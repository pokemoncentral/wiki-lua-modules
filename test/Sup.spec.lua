-- "Test cases" for Sup

local lib = require("test/snapshot-lib")
local sup = require("Sup")

lib.printHeading("A very long one")
local solid = sup.OACRFVFHGSS({ args = {} })
print(solid)

lib.printHeading("The very long one, but split")
local split = sup.OA({ args = { "C RFVF", "HGSS" } })
print(split)

-- Syntax equivalence

assert(solid == split, "syntax mismatch")

lib.printHeading("No color specified")
print(sup.LGPE({ args = {} }))

lib.printHeading("_abbr function")
print(sup._abbr({ args = { "RBG RFVF", "LGPE" } }))
