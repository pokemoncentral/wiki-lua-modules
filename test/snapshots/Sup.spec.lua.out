<sup>[[Pokémon Oro e Argento|<span style="color: #DAA520;">'''O'''</span><span style="color: #C0C0C0;">'''A'''</span>]][[Pokémon Cristallo|<span style="color: #4FD9FF;">'''C'''</span>]][[Pokémon Rosso Fuoco e Verde Foglia|<span style="color: #FF7327;">'''RF'''</span><span style="color: #00DD00;">'''VF'''</span>]][[Pokémon Oro HeartGold e Argento SoulSilver|<span style="color: #B69E00;">'''HG'''</span><span style="color: #C0C0E1;">'''SS'''</span>]]</sup>
<sup>[[Pokémon Oro e Argento|<span style="color: #DAA520;">'''O'''</span><span style="color: #C0C0C0;">'''A'''</span>]][[Pokémon Cristallo|<span style="color: #4FD9FF;">'''C'''</span>]][[Pokémon Rosso Fuoco e Verde Foglia|<span style="color: #FF7327;">'''RF'''</span><span style="color: #00DD00;">'''VF'''</span>]][[Pokémon Oro HeartGold e Argento SoulSilver|<span style="color: #B69E00;">'''HG'''</span><span style="color: #C0C0E1;">'''SS'''</span>]]</sup>
<sup>[[Pokémon: Let's Go, Pikachu! e Let's Go, Eevee!|<span style="color: #000000;">'''LG'''</span><span style="color: #ffcf11;">'''P'''</span><span style="color: #c47e39;">'''E'''</span>]]</sup>
<sup>[[Pokémon Rosso e Blu|<span style="color: #FF1111;">'''R'''</span><span style="color: #1111FF;">'''B'''</span>]][[Pokémon Giallo|<span style="color: #FFD733;">'''G'''</span>]][[Pokémon Rosso Fuoco e Verde Foglia|<span style="color: #FF7327;">'''RF'''</span><span style="color: #00DD00;">'''VF'''</span>]][[Pokémon: Let's Go, Pikachu! e Let's Go, Eevee!|<span style="color: #000000;">'''LG'''</span><span style="color: #ffcf11;">'''P'''</span><span style="color: #c47e39;">'''E'''</span>]]</sup>
